File: src/main/kotlin/br/com/home/awsmicroservices/repository/ProductRepository.java
package br.com.home.awsmicroservices.repository;

import br.com.home.awsmicroservices.model.Product;
import org.springframework.data.repository.CrudRepository;

import java.util.Optional;

public interface ProductRepository extends CrudRepository<Product, Long> {

    Optional<Product> findByCode(String code);

}

File: src/main/kotlin/br/com/home/awsmicroservices/repository/ProductRepository.kt
package br.com.home.awsmicroservices.repository

import br.com.home.awsmicroservices.model.Product
import org.springframework.data.repository.CrudRepository
import java.util.*

interface ProductRepository : CrudRepository<Product, Long> {
    fun findByCode(code: String): Optional<Product>
}